File: tests/go/tests-go-testing-allure3-playwright/tests/infra/layout_css_test.go
package infra_test

import (
	"testing"

	tests "tests-go-testing-allure3-playwright"

	allure "github.com/allure-framework/allure-go/commons/gotest"
	"github.com/stretchr/testify/require"
)

func wrapLayout(t *testing.T, name string, body func(*allure.Context)) {
	t.Helper()
	tests.Wrap(t, name, body, tests.LayerInfraFrontend("LayoutCss", "Test infra", "Layout CSS", "normal")...)
}

func TestGridColumnCountParsesGridTemplateColumns(t *testing.T) {
	wrapLayout(t, "gridColumnCount parses grid-template-columns", func(a *allure.Context) {
		require.Equal(t, 3, tests.GridColumnCount("repeat(3, minmax(0, 1fr))"))
		require.Equal(t, 2, tests.GridColumnCount("603px 603px"))
		require.Equal(t, 1, tests.GridColumnCount("1fr"))
		require.Equal(t, 1, tests.GridColumnCount("316px"))
		require.Equal(t, 0, tests.GridColumnCount("none"))
		require.Equal(t, 0, tests.GridColumnCount(""))
		require.Equal(t, 0, tests.GridColumnCount("   "))
	})
}

File: tests/csharp/tests-csharp-xunit-api_request-playwright/Tests/Infra/LayoutCssTest.cs
using Allure.Net.Commons;
using Allure.Net.Commons.Attributes;
using Helpers;
using Tests;

namespace Tests.Infra;

[AllureLabel("layer", "infra")]
[AllureEpic("Test infra")]
[AllureFeature("Layout CSS")]
[AllureSeverity(SeverityLevel.normal)]
[Trait("TestCategory", "infra")]
[Trait("TestCategory", "infra_frontend")]
[AllureSuite("LayoutCss")]
public sealed class LayoutCssTest : AllureMeta
{
    [InlineData("repeat(3, minmax(0, 1fr))", 3)]
    [InlineData("603px 603px", 2)]
    [InlineData("1fr", 1)]
    [InlineData("316px", 1)]
    [InlineData("none", 0)]
    [InlineData(null, 0)]
    [InlineData("", 0)]
    [InlineData("   ", 0)]
    [Theory(DisplayName = "gridColumnCount parses grid-template-columns")]
    public void GridColumnCountParsesGridTemplateColumns(string? gridTemplateColumns, int expected)
    {
        Assert.Equal(expected, LayoutCss.GridColumnCount(gridTemplateColumns));
    }
}
